### PR TITLE
[ENG-641] - Add checklist component

### PR DESCRIPTION
**Pull request by floreks**
Fixes: [ENG-641](https://linear.app/pluralsh/issue/ENG-641/create-checklist-component) | [ENG-666](https://linear.app/pluralsh/issue/ENG-666/implement-tertiary-no-padding-button-variant) | [ENG-642](https://linear.app/pluralsh/issue/ENG-642/add-motion-design-to-checklist)

[Figma](https://www.figma.com/file/EoXLxyyRIe6jfyjKegZo1Y/Onboarding-Checklist?node-id=46%3A6649)

### CODE REVIEW

**Comment by maciaszczykm** [comment]
@floreks Preview is failing at the moment.

**Review by maciaszczykm — APPROVED** [review]
Looks amazing, especially with all animations on transitions. One question, should this focused state be rendered while opening it for the first time (I didn't click anything)?

Zrzut ekranu 2022-10-7 o 12 13 21

**Comment by floreks** [comment]
I did it on purpose actually so that it is immediately possible to use keyboard for navigation. It can be easily changed if needed.

File: src/components/Checklist.tsx [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+  children: ReactElement<ChecklistItemProps>[]
+}
+
+type ChecklistStateProps = DOMProps & {
+  onSelectionChange?: Dispatch<number>
+  selectedKey?: number,
+  completedKey?: number,
+  isOpen?: boolean
+  isDismissed?: boolean
+}
```
Comment (dogmar) [inline src/components/Checklist.tsx:84]: What is  `DOMProps` for? It doesn't seem like you're using any other props than the ones you've explicitly specified.

File: src/components/ChecklistItem.tsx [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+import {
+  ComponentPropsWithRef, Dispatch, ReactElement, useEffect, useRef, useState,
+} from 'react'
+import AnimateHeight from 'react-animate-height'
+import styled from 'styled-components'
+
+import CaretDownIcon from './icons/CaretDownIcon'
+import SuccessIcon from './icons/SuccessIcon'
+
+const heightAnimationDuration = 666 // 666ms
```
Comment (dogmar) [inline src/components/ChecklistItem.tsx:11]: Interesting choice of number. 🤔
Though I do think the animation feels a little slow. I'd consider speeding this up to something more like 200-500. Just a personal opinion though.

**Review by dogmar — COMMENTED** [review]
This looks great. Very smooth interactions, and I love the circular animation of the borders of the step numbers.

**Review by dogmar — APPROVED** [review]

File: src/components/Checklist.tsx (hunk 1 above)
Comment (floreks) [reply src/components/Checklist.tsx:84]: Good catch. That's just a leftover. I'll remove it.

File: src/components/ChecklistItem.tsx (hunk 2 above)
Comment (floreks) [reply src/components/ChecklistItem.tsx:11]: I have decreased some animation times to make them faster. Indeed it looks smoother.

**Pull request merged**